qiskit/primitives/base_result.py

```diff
@@ -0,0 +1,87 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Primitive result abstract base class
+"""
+
+from __future__ import annotations
+
+from abc import ABC
+from collections.abc import Iterator, Sequence
+from dataclasses import fields
+from typing import Any, Dict
+
+from numpy import ndarray
+
+
+ExperimentData = Dict[str, Any]
+
+
+class BasePrimitiveResult(ABC):
+    """Primitive result abstract base class.
+
+    Base class for Primitive results meant to provide common functionality to all inheriting
+    result dataclasses.
+    """
+
+    def __post_init__(self) -> None:
+        """
+        Verify that all fields in any inheriting result dataclass are consistent, after
+        instantiation, with the number of experiments being represented.
+
+        This magic method is specific of `dataclasses.dataclass`, therefore all inheriting
+        classes must have this decorator.
+
+        Raises:
+            TypeError: If one of the data fields is not a Sequence or ``numpy.ndarray``.
+            ValueError: Inconsistent number of experiments across data fields.
+        """
+        for value in self._field_values:  # type: Sequence
+            # TODO: enforce all data fields to be tuples instead of sequences
+            if not isinstance(value, (Sequence, ndarray)) or isinstance(value, str):
```

Comment:
I'm wondering whether the check `isinstance(value, str)` is enough or not. What if `bytes` object is given?

Reply:
Should we check each entry of `Sequence` perhaps? Do you have any suggestion, @ikkoham?

Reply:
The problem with checking each sequence entry is that the types will differ for each data attribute (e.g. values and metadata), and between primitives (e.g. sampler and estimator). So those checks should probably live under the particular primitives `__post_init__` method, instead of inside `BasePrimitiveResult`.

Is this what you are referring to?

Reply:
I see. I just wondered whether there is any simpler way of this check.